python/tank/util/shotgun.py

```diff
@@ -965,6 +965,8 @@ def register_publish(tk, context, path, name, version_number, **kwargs):
 
         - ``sg_fields`` - Some additional Shotgun fields as a dict (e.g. ``{'tag_list': ['foo', 'bar']}``)
 
+        - ``dry_run`` - Boolean. If set, do not actually create a database entry. Return the
```

Comment:
please outline the differences between the return data between a dry run pub and a proper one!